### PR TITLE
k3s: use team for maintainers

### PR DESCRIPTION
**Pull request by superherointj**
k3s: use team for maintainers

- eliminates duplication of places to log maintainers.
- adds missing entries of upkeep.

For awareness, all K3s maintainers have maintainer privileges at [teams](https://github.com/orgs/NixOS/teams/k3s). Which enables to manipulate the maintainers list as needed to sync [team-list.nix](https://github.com/NixOS/nixpkgs/blob/master/maintainers/team-list.nix).

CC @marcusramberg @yajo @Mic92 @euank @wrmilling

### CODE REVIEW

**Review by marcusramberg — APPROVED** [review]
Lgtm!

File: nixos/modules/services/cluster/k3s/default.nix [hunk 1]
```diff
@@ -432,4 +432,6 @@ in
       };
     };
   };
+
+  meta.maintainers = [ lib.teams.k3s.members ];
```
Comment (yajo) [inline nixos/modules/services/cluster/k3s/default.nix:436]: A list of lists is supported? Should'nt it be like this?

```suggestion
  meta.maintainers = lib.teams.k3s.members;
```

Same below

Comment (superherointj) [reply nixos/modules/services/cluster/k3s/default.nix:436]: `meta.maintainers` is a list. I also made that mistake just a few minutes ago. :-)

Comment (superherointj) [reply nixos/modules/services/cluster/k3s/default.nix:436]: Yeah, I need to double check this. It was that way before. But was erroring in nixfmt.

Comment (superherointj) [reply nixos/modules/services/cluster/k3s/default.nix]: Thanks yajo!

**Review by yajo — APPROVED** [review]

**Review by wrmilling — APPROVED** [review]

**Pull request merged**